src/benchmark.py

```diff
@@ -0,0 +1,31 @@
+from jina import Document, DocumentArray
+from timecontext import TimeContext
+
+
+def benchmark(cls, function, *args, **kwargs):
+    def wrapper(*args, **kwargs):
+        with TimeContext() as timer:
+            func = function(*args, **kwargs)
+
+        cls.profile[f'{function.__name__}']['time'] += timer.duration
+        cls.profile[f'{function.__name__}']['calls'] += 1
+        return func
+
+    return wrapper
+
+
+class BenchmarkedDocument(Document):
```

Comment:
Shouldn't this be a simple function instead?
```suggestion
def BenchmarkedDocument():
```